### PR TITLE
Geo classifier

### PR DESCRIPTION
**Pull request by p0rtale**
С хардкодом и валидацией на трейне :sunglasses:

### CODE REVIEW

File: geo_classifier/dataset.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            self.map_label_country = json.load(fin)
+
+        self.dataset = pd.read_csv(os.path.join(self.path, "datasets/dataset_label.csv"))
+
+    def __len__(self):
+        return len(self.dataset * 3)
+
+    def __getitem__(self, idx) -> Tuple[torch.Tensor, dict]:
+        subimg_num = idx % 3
+        data = self.dataset.iloc[idx // 3]
```
Comment (kaizer-nurik) [inline geo_classifier/dataset.py:31]: молодца

File: geo_classifier/train.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+        return model
+
+    def forward(self, x):
+        return self.model(x)
+
+    def configure_optimizers(self):
+        optim_feature_extrator = torch.optim.SGD(
+            self.parameters(), **self.hparams.optim["params"]
+        )
```
Comment (kaizer-nurik) [inline geo_classifier/train.py:48]: SGD? 🤔

File: geo_classifier/train.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    def __build_model(self, freeze):
+        logging.info("Build model")
+
+        model = torchvision.models.resnet50(weights='ResNet50_Weights.DEFAULT')
+
+        if freeze:
+            for param in model.parameters():
+                param.requires_grad = False
+
+        model.fc = torch.nn.Linear(model.fc.in_features, self.hparams.countries_num)
```
Comment (kaizer-nurik) [inline geo_classifier/train.py:38]: 🤔 Достаточно одного линейного слоя? и без софтмакса?

Comment (kaizer-nurik) [reply geo_classifier/train.py:38]: https://stackoverflow.com/questions/55675345/should-i-use-softmax-as-output-when-using-cross-entropy-loss-in-pytorch softmax не нужен

File: geo_classifier/train.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        self.val_acc(torch.argmax(preds, dim=1), targets)
+
+        self.log("val_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        self.log("val_acc", self.val_acc, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+
+    def train_dataloader(self):
+        transforms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.RandomHorizontalFlip(),
+                torchvision.transforms.RandomResizedCrop(224, scale=(0.66, 1.0)),
```
Comment (kaizer-nurik) [inline geo_classifier/train.py]: Зачем?

File: geo_classifier/train.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        return dataloader
+
+    def val_dataloader(self):
+        transforms = torchvision.transforms.Compose(
+            [
+                torchvision.transforms.Resize(256),
+                torchvision.transforms.CenterCrop(224),
+                torchvision.transforms.ToTensor(),
+                torchvision.transforms.Normalize(
+                    (0.485, 0.456, 0.406), (0.229, 0.224, 0.225)
```
Comment (kaizer-nurik) [inline geo_classifier/train.py:117]: https://stackoverflow.com/questions/58151507/why-pytorch-officially-use-mean-0-485-0-456-0-406-and-std-0-229-0-224-0-2

**Review by kaizer-nurik — APPROVED** [review]
